test/e2e/specs/editor/blocks/buttons.spec.js

```diff
@@ -0,0 +1,136 @@
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'has focus on button content', async ( { editor, page } ) => {
+		await editor.insertBlock( { name: 'core/buttons' } );
+		await page.keyboard.type( 'Content' );
+
+		//Check the content
+		const content = await editor.getEditedPostContent();
+		expect( content ).toBe(
+			`<!-- wp:buttons -->
+<div class="wp-block-buttons"><!-- wp:button -->
+<div class="wp-block-button"><a class="wp-block-button__link wp-element-button">Content</a></div>
+<!-- /wp:button --></div>
+<!-- /wp:buttons -->`
+		);
+	} );
+
+	test( 'has focus on button content (slash inserter)', async ( {
+		editor,
+		page,
+	} ) => {
+		await page.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( '/buttons' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'Content' );
+
+		//Check the content
+		const content = await editor.getEditedPostContent();
+		expect( content ).toBe(
+			`<!-- wp:buttons -->
+<div class="wp-block-buttons"><!-- wp:button -->
+<div class="wp-block-button"><a class="wp-block-button__link wp-element-button">Content</a></div>
+<!-- /wp:button --></div>
+<!-- /wp:buttons -->`
+		);
+	} );
+
+	test( 'dismisses link editor when escape is pressed', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		// Regression: https://github.com/WordPress/gutenberg/pull/19885
+		await editor.insertBlock( { name: 'core/buttons' } );
+		await pageUtils.pressKeyWithModifier( 'primary', 'k' );
+		await page.waitForFunction(
+			() => !! document.activeElement.closest( '.block-editor-url-input' )
+		);
+		await page.keyboard.press( 'Escape' );
+		await page.waitForFunction(
+			() =>
+				document.activeElement ===
+				document.querySelector( '.block-editor-rich-text__editable' )
+		);
+		await page.keyboard.type( 'WordPress' );
+
+		// Check the content
+		const content = await editor.getEditedPostContent();
+		expect( content ).toBe(
+			`<!-- wp:buttons -->
+<div class="wp-block-buttons"><!-- wp:button -->
+<div class="wp-block-button"><a class="wp-block-button__link wp-element-button">WordPress</a></div>
+<!-- /wp:button --></div>
+<!-- /wp:buttons -->`
+		);
+	} );
+
+	test( 'moves focus from the link editor back to the button when escape is pressed after the URL has been submitted', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		// Regression: https://github.com/WordPress/gutenberg/issues/34307
+		await editor.insertBlock( { name: 'core/buttons' } );
+		await pageUtils.pressKeyWithModifier( 'primary', 'k' );
+		await page.waitForFunction(
+			() => !! document.activeElement.closest( '.block-editor-url-input' )
+		);
+		await page.keyboard.type( 'https://example.com' );
+		await page.keyboard.press( 'Enter' );
+		await page.waitForFunction(
+			() =>
+				document.activeElement ===
+				document.querySelector(
+					'.block-editor-link-control a[href="https://example.com"]'
+				)
+		);
```

Comment:
Ditto.

```suggestion
		await expect( page.locator( 'role=link[name=/^https:\\/\\/example\\.com]' ) ).toBeFocused();
```

Reply:
Hi @kevin940726, Tried to change the selector as per suggestion but it is showing syntax error in regex. Tried to modify it but it is still showing parsing error. Could you please check on this?

![image](https://user-images.githubusercontent.com/41000648/172535129-a130393b-3180-45b9-bbd3-a2c18ddc3196.png)

Reply:
Ahh, yes, I was missing the `/` in the end.

```suggestion
		await expect( page.locator( 'role=link[name=/^https:\\/\\/example\\.com/]' ) ).toBeFocused();
```

Reply:
@kevin940726 

Updated, but still seems some issue, am I missing something?

![image](https://user-images.githubusercontent.com/41000648/172545614-dfa3dde1-335d-4523-aa48-2ed739c60a11.png)

Reply:
Hmm, not sure! Could you push the changes to this branch so that I can take a look?

Reply:
Hi,
I have pushed the code but the CI failed on same line.

<img width="1066" alt="Screenshot 2022-06-16 at 12 23 46 PM" src="https://user-images.githubusercontent.com/41000648/174010008-01daaf76-5bb5-4151-89ab-7223bb36d17a.png">

--

Can we use this 👇  selector instead?

```
await expect(
      page.locator( 'a[href="https://example.com/"]' )
).toBeFocused();
```

Reply:
Oh, I think the URL input automatically trims the `https://` part off, so it's just `wordpress.org` now.

I think we can do this instead:

```js
await expect( page.locator( 'role=link[name=/^example\\.com/]' ) ).toBeFocused();
```

Reply:
Addressed the feedbacks.